cpp/src/feather/metadata.cc

```diff
@@ -326,11 +325,9 @@ class ColumnBuilder::Impl {
   ColumnType::type type_;
 
   // Type-specific metadata union
-  union {
```

Comment:
If we really do want a union here (since modern compilers seem to accept it in C++11 mode) we might just want to scope the change made here to older gcc.